### DISCUSSION TITLE
Define "root"

### DISCUSSION THREAD

**Discussion by KieranCarden** [maktoobgar/scene_manager#34] [Ideas]
Not sure how feasible this would be, or if it would be worth exploring, but as of right now when changing levels/scenes it replaces at the root level, so everything has to exist within the scene that's being replaced (player, etc).

I'm looking to build a sort of "base" at the root that holds everything else in separated sections:
![image](https://github.com/user-attachments/assets/74e04bb3-4319-4a61-ae74-cdc1da2efe6f)

![image](https://github.com/user-attachments/assets/946a2389-6a22-4737-8c76-da47620775ae)

Right now transitioning from "Start" to a level will replace all of "BaseGame". My curiosity is wondering if it would be worthwhile to allow being able to define the top level that SceneManager would then use as the "root", or if not defined continue functioning as-is. In my case it would be "LevelBase" and everything under that would get replaced where I could then retain everything outside of it.

**Comment by maktoobgar** [comment]
Hello,

Good idea, this is actually quite possible. Some functions need some small changes and it works fine.
As soon as I get the time for it, I'll be on it.
Thanks for the idea. 🙌
